packages/@ember/-internals/runtime/lib/system/array_proxy.js

```diff
@@ -119,7 +120,14 @@ export default class ArrayProxy extends EmberObject {
       // revalidate eagerly if we're being tracked, since we no longer will
       // be able to later on due to backtracking re-render assertion
       this._revalidate();
-      return combine(getChainTagsForKey(this, `arrangedContent.${key}`, addMandatorySetter));
+      let tags = [this._arrangedContentTag];
+      let arrangedContent = get(this, 'arrangedContent');
+
+      if (isObject(arrangedContent)) {
+        tags.push(tagForProperty(arrangedContent, key, addMandatorySetter));
+      }
```

Comment:
in the else here there is no point in it being an array and calling combine, why not move the combine inside here